src/nitcatalog.nit

```diff
@@ -0,0 +1,815 @@
+		end
+		res.add "</ul>\n"
+
+		res.add "<h3>Tags</h3>\n"
+		var tags = mproject.metadata("project.tags")
+		var ts2 = new Array[String]
+		var cat = null
+		if tags != null then
+			var ts = tags.split(",")
+			for t in ts do
+				t = t.trim
+				if t == "" then continue
+				if cat == null then cat = t
+				tag2proj[t].add mproject
+				t = t.html_escape
+				ts2.add "<a href=\"index.html#tag_{t}\">{t}</a>"
+			end
+			res.add_list(ts2, ", ", ", ")
+		end
+		if ts2.is_empty then
+			var t = "none"
+			cat = t
+			tag2proj[t].add mproject
+			res.add "<a href=\"index.html#tag_{t}\">{t}</a>"
+		end
+		if cat != null then cat2proj[cat].add mproject
+		score += ts2.length.score
+
+		var reqs = deps[mproject].greaters.to_a
+		reqs.remove(mproject)
+		alpha_comparator.sort(reqs)
+		res.add "<h3>Requirements</h3>\n"
+		if reqs.is_empty then
+			res.add "none"
+		else
+			var list = new Array[String]
+			for r in reqs do
+				var direct = deps.has_direct_edge(mproject, r)
+				var s = "<a href=\"{r}.html\">"
+				if direct then s += "<strong>"
+				s += r.to_s
+				if direct then s += "</strong>"
+				s += "</a>"
+				list.add s
+			end
+			res.add_list(list, ", ", " and ")
+		end
+
+		reqs = deps[mproject].smallers.to_a
+		reqs.remove(mproject)
+		alpha_comparator.sort(reqs)
+		res.add "<h3>Clients</h3>\n"
+		if reqs.is_empty then
+			res.add "none"
+		else
+			var list = new Array[String]
+			for r in reqs do
+				var direct = deps.has_direct_edge(r, mproject)
+				var s = "<a href=\"{r}.html\">"
+				if direct then s += "<strong>"
+				s += r.to_s
+				if direct then s += "</strong>"
+				s += "</a>"
+				list.add s
+			end
+			res.add_list(list, ", ", " and ")
+		end
+
+		score += deps[mproject].greaters.length.score
+		score += deps[mproject].direct_greaters.length.score
+		score += deps[mproject].smallers.length.score
+		score += deps[mproject].direct_smallers.length.score
+
+		var contributors = mproject.contributors
+		if not contributors.is_empty then
+			res.add "<h3>Contributors</h3>\n<ul class=\"box\">"
+			for c in contributors do
+				add_contrib(c, mproject, res)
+			end
+			res.add "</ul>"
+		end
+		score += contributors.length.to_f
+
+		var mmodules = 0
+		var mclasses = 0
+		var mmethods = 0
+		var loc = 0
+		for g in mproject.mgroups do
+			mmodules += g.module_paths.length
+			for m in g.mmodules do
+				var am = modelbuilder.mmodule2node(m)
+				if am != null then
+					var file = am.location.file
+					if file != null then
+						loc += file.line_starts.length - 1
+					end
+				end
+				for cd in m.mclassdefs do
+					mclasses += 1
+					for pd in cd.mpropdefs do
+						if not pd isa MMethodDef then continue
+						mmethods += 1
+					end
+				end
+			end
+		end
+		self.mmodules[mproject] = mmodules
+		self.mclasses[mproject] = mclasses
+		self.mmethods[mproject] = mmethods
+		self.loc[mproject] = loc
+
+		#score += mmodules.score
+		score += mclasses.score
+		score += mmethods.score
+		score += loc.score
```

Comment:
Some the number of lines of code helps to be in the top 10? And why not the modules?